lib/devise_two_factorable/models/two_factorable.rb

```diff
@@ -118,6 +123,12 @@ def validate_otp_recovery_token(token)
 
     def validate_otp_token_with_drift(token)
       # should be centered around saved drift
+      if self.last_successful_otp_at
```

Comment:
Could you extract the conditional to a method and put it as a one-liner guard clause? Like:
```ruby
def drift_window_overlap?
  (self.last_successful_otp_at - Time.now).abs < self.class.otp_drift_window
end

def last_successful_opt_at_past?
  self.last_successful_otp_at && drift_window_overlap?
end

def validate_otp_token_with_drift(token)
  return false if last_successful_opt_at_past?
  # .......etc
  # ....etc
end
```

Reply:
Yeah, let's get this refactored a bit before submitting
I'm not certain @cchoi12's example does what we intend, but just about any nested if should be avoidable.